.github/workflows/_docker-test.yml

```diff
@@ -0,0 +1,104 @@
+name: Reusable docker image test workflow
+
+on:
+  workflow_call:
+    inputs:
+      build-args:
+        description: "List of build-time variables."
+        required: false
+        type: string
+      ref:
+        required: false
+        type: string
+    secrets:
+      dockerhub-user:
+        required: true
+      dockerhub-token:
+        required: true
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Test image
+    runs-on: "ubuntu-22.04"
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
What's the benefit of using a hash instead of the version (as we've previously used in other CI steps)? Is it so that we can use a minor+patched version?

Reply:
I also see this repeated in other steps like for `docker@setup-buildx-action` and `Wandalen/wretry.action`.